layouts/posts/li.html

```diff
@@ -0,0 +1,4 @@
+<li>
+  <span>{{ .Date.Format (.Site.Params.dateFormat | default "January 2, 2006" ) }}</span>
+  <a href="{{ .URL }}">{{ .Title }}</a>
+</li>
```

Comment:
I don't think we should split this one if we aren't using the code in two or more places.

Reply:
True, it might only be used once, but that's the whole point of this. `li` here is one _view_ of the `post` type, potentially in the future we might have `summary`, `short`, `expanded`, etc as other views for post.

Essentially these views are standalone representation of that particular type.

Reply:
Understood.
So, Does not make sense to move this file to `partials/post/`?

Reply:
Also, why you used these constructions with `{-` and `-}`? Never saw it before. Can you share any documentation about it?

Reply:
Not always, this is the difference between "view of something" and "reusable piece of code". For example there's `single` view (which is a default known one to Hugo, but a view nonetheless) is defined outside of partial on its own directory but uses partial (`disqus` and `math`) which are not _view_ of any type.

Reply:
Those `{{-` and `-}}` and left and right white-space trimming in go template ([here](https://golang.org/pkg/text/template/#hdr-Text_and_spaces)).

Go processes everything in `{{` and `}}` and retains white space and carriage return, so one line of html code inside a multi-line of go _logic_ will have multiple extra empty lines around it (such as (e.g. empty spaces around `<link hre="coder.css" />`)

It looks nicer when viewing source of the generated page without extra empty spaces, when not required.